src/services/PostEventProcessor.ts

```diff
@@ -0,0 +1,133 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+
+import { AppError } from "../utils/AppError";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { Constants } from "../utils/Constants";
+import { IPRService } from "./IPRService";
+import { EnvironmentVariables } from "./EnvironmentVariables";
+import { ServicesEnum } from "../models/enums/ServicesEnum";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { IpvStartedOnEvent, JourneyWentAsyncOnEvent, ReadyToResumeOnEvent, AccountDeletedOnEvent } from "../models/IPREventTypes";
+
+
+export class PostEventProcessor {
+	private static instance: PostEventProcessor;
+
+	private readonly logger: Logger;
+
+	private readonly metrics: Metrics;
+
+	private readonly environmentVariables: EnvironmentVariables;
+
+	private readonly iprService: IPRService;
+
+	constructor(logger: Logger, metrics: Metrics) {
+		this.logger = logger;
+		this.metrics = metrics;
+		this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.POST_EVENT_SERVICE);
```

Comment:
Given how much login we have in the `EnvironmentVariables`, is now a good time to suggest we Split it based on service? 
We might want to introduce an EnvironmentVariable interface for core properties like
```
process.env.POWERTOOLS_LOG_LEVEL 
process.env.POWERTOOLS_METRICS_NAMESPACE
```
that every service has, but allow for service specific implementations. eg. 

```
class PostEventConfiguration extends CoreConfiguration {
constructor(source) {...}
SERVICE_NAME = 'PostEvent'
...
}
```

Reply:
As mentioned by Viveak, the environmentVariables code will be refactored using another ticket.